lib/core/include/scipp/core/sizes.h

```diff
@@ -15,6 +17,7 @@
 namespace scipp::core {
 
 constexpr int32_t NDIM_MAX = 6;
+constexpr int32_t NDIM_STACK = 4;
```

Comment:
Why is this smaller than `NDIM_MAX`?

Reply:
Partially to see if we have anything that unintentionally relies on them being the same. But even once we know I think I'd keep it at 4, since having more than 4 dimensions is very rare.